In [ ]:
import pandas as pd 
from scipy.stats import pointbiserialr
from scipy.stats import pearsonr
import matplotlib.pyplot as plt

In [ ]:
df = pd.read_csv('general_data.csv')

In [ ]:
df.head()

Age Attrition  ... YearsSinceLastPromotion YearsWithCurrManager
0   51        No  ...                       0                    0
1   31       Yes  ...                       1                    4
2   32        No  ...                       0                    3
3   38        No  ...                       7                    5
4   32        No  ...                       0                    4

[5 rows x 24 columns]

In [ ]:
df.columns

Index(['Age', 'Attrition', 'BusinessTravel', 'Department', 'DistanceFromHome',
       'Education', 'EducationField', 'EmployeeCount', 'EmployeeID', 'Gender',
       'JobLevel', 'JobRole', 'MaritalStatus', 'MonthlyIncome',
       'NumCompaniesWorked', 'Over18', 'PercentSalaryHike', 'StandardHours',
       'StockOptionLevel', 'TotalWorkingYears', 'TrainingTimesLastYear',
       'YearsAtCompany', 'YearsSinceLastPromotion', 'YearsWithCurrManager'],
      dtype='object')

In [ ]:
df.Attrition.replace(to_replace = ('No','Yes'),value = (0,1),inplace = True)

In [ ]:
stats, p = pearsonr(df.Attrition, df.Age)
print(stats, p)

-0.15920500686577949 1.996801615887198e-26


#Hypothesis 1
null hypothesis (H0) = Age and attrition have no correlation .

alternate hypothesis (H1) = Age and attrition have a correlation between them .

r = -0.15920500686577965

p = 1.996801615886744e-26

r is negative so there is a negative relation between Age and attrition

and as p <0.05 so null hypothesis (H0) is rejected and we find that there is a negative correlation between attrition and age

In [ ]:
stats, p = pearsonr(df.Attrition, df.DistanceFromHome)
print(stats, p)

-0.009730141010179692 0.5182860428050771


#Hypothesis 2
null hypothesis (H0) = Attrition and distance from home have no correlation .

alternate hypothesis (H1) = Attrition and distance from home have a correlation between them .

r = -0.00973014101017967

p = 0.5182860428050771

Here p is greater than 0.05 hence we accept the null hypothesis (H0) that there is no realtion between attrtion and distance from home

In [ ]:
stats, p = pearsonr(df.Attrition, df.MonthlyIncome)
print(stats, p)

-0.03117628169811501 0.03842748490600132


#Hypothesis 3
null hypothesis (H0) = Attrition and monthly income have no correlation .

alternate hypothesis (H1) = Attrition and monthly income have a correlation between them .

r = -0.03117628169811501

p = 0.03842748490600132

As p is lesser than .05 we reject null hypothesis (H0) and conclude that there is a significant relation between TRRITION AND MONTHLY INCOME.

r is -.0311 so we see there is negative relation between the two

In [ ]:
stats, p = pearsonr(df.Attrition, df.PercentSalaryHike)
print(stats, p)

0.0325325948910535 0.030743386433355353


#Hypothesis 4
null hypothesis (H0) = Attrition and salary hike percentage have no correlation .

alternate hypothesis (H1) = Attrition and salary hike percentage have a correlation between them .

r = 0.03253259489105349

p = 0.030743386433355353

As p is lesser than .05 we reject null hypothesis (H0) and conclude that there is a significant relation between Attrition and salary hike percent .

r is 0.032 so we see there is positive relation between the two

In [ ]:
stats, p = pearsonr(df.Attrition, df.TrainingTimesLastYear)
print(stats, p)

-0.049430576244254974 0.0010247061915365072


#Hypothesis 5
null hypothesis (H0) = Attrition and TrainingTimesLastYear have no correlation .

alternate hypothesis (H1) = Attrition and TrainingTimesLastYear have a correlation between them .

r = -0.049430576244255

p = 0.0010247061915365072

As p is lesser than .05 we reject null hypothesis (H0) and conclude that there is a significant relation between Attrition and TrainingTimesLastYear .

r is -0.049 so we see there is negative relation between the two

In [ ]:
stats, p = pearsonr(df.Attrition, df.YearsAtCompany)
print(stats, p)

-0.13439221398997717 3.1638831224877484e-19



#Hypothesis 6
null hypothesis (H0) = Attrition and YearsAtCompany have no correlation .

alternate hypothesis (H1) = Attrition and YearsAtCompany have a correlation between them .

r = -0.1343922139899772

p = 3.1638831224877484e-19

As p is lesser than .05 we reject null hypothesis (H0) and conclude that there is a significant relation between Attrition and YearsAtCompany .

r is -0.134 so we see there is negative relation between the two

In [ ]:
stats, p = pearsonr(df.Attrition, df.YearsSinceLastPromotion)
print(stats, p)

-0.03301877514258437 0.028330336189396753


#Hypothesis 7
null hypothesis (H0) = Attrition and YearsSinceLastPromotion have no correlation .

alternate hypothesis (H1) = Attrition and YearsSinceLastPromotion have a correlation between them .

r = -0.03301877514258433

p = 0.028330336189396753

As p is lesser than .05 we reject null hypothesis (H0) and conclude that there is a significant relation between Attrition and YearsSinceLastPromotion .

r is -0.033 so we see there is negative relation between the two

In [ ]:
stats, p = pearsonr(df.Attrition, df.YearsWithCurrManager)
print(stats, p)

-0.15619931590162806 1.7339322652900218e-25


#Hypothesis 8
null hypothesis (H0) = Attrition and YearsWithCurrManager have no correlation .

alternate hypothesis (H1) = Attrition and YearsWithCurrManager have a correlation between them .

-0.15619931590162847

1.7339322652896276e-25

As p is lesser than .05 we reject null hypothesis (H0) and conclude that there is a significant relation between Attrition and YearsWithCurrManager.

r is -0.156 so we see there is negative relation between the two